File: pyStudy/day1227/Study03.java
package day1227;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class Study03 {
    public static void main(String[] args) {
        List<Object> a = new ArrayList<>(Arrays.asList(1,2,3));
        a.add(4);
        System.out.println(a);

        a.add(Arrays.asList(5, 6, 7));
        System.out.println(a);

        List<Integer> b = new ArrayList<>(Arrays.asList(1,4,3,2));
        System.out.println(b);
        Collections.sort(b);
        System.out.println(b);
        b.sort(Collections.reverseOrder());
        System.out.println(b);

        List<String> str = new ArrayList<>(Arrays.asList("a","c","b","e"));
        Collections.sort(str);
        System.out.println(str);

        List<String> c = new ArrayList<>(Arrays.asList("a","c","b","e"));
        Collections.reverse(c);
        System.out.println(c);

        List<Integer> d = new ArrayList<>(Arrays.asList(1,2,3));
        System.out.println(d.indexOf(3));
        
        d.add(1,4);
        System.out.println(d);

        List<Integer> e = new ArrayList<>(Arrays.asList(1,2,3,1,2,3));
        e.remove((Integer)3);
        System.out.println(e);
        e.remove((Integer)3);
        System.out.println(e);

        List<Integer> f = new ArrayList<>(Arrays.asList(1,2,3));
        System.out.println(f.remove(f.size()-1));
        System.out.println(f);
        
        List<Integer> g = new ArrayList<>(Arrays.asList(1,2,3,4,5));
        System.out.println(g.remove(1));
        System.out.println(g);

        List<Integer> h = new ArrayList<>(Arrays.asList(1,2,3,1));
        System.out.println(Collections.frequency(h, 1));

        List<Integer> i = new ArrayList<>(Arrays.asList(1,2,3));
        i.addAll(Arrays.asList(4,5));
        System.out.println(i);

    }
}

File: pyStudy/day1227/study03.py
# 리스트 관련 함수

# 추가하기
a = [1,2,3];
a.append(4);
print(a);
a.append([5,6,7]);
print(a);

# 리스트 정렬(sort)
a = [1,4,3,2];
a.sort()                # 오름차순
print(a);
a.sort(reverse=True)    # 내림차순
print(a)

str = ['a','c','b'];
print(str);
str.sort();
print(str);

# 리스트 정렬(reverse) - 뒤집기
a = ['a','c','b','e'];
a.reverse();
print(a);

# 인덱스 반환
a = [1,2,3];
print(a.index(2));

# 리스트 요소 삽입
a = [1,2,3,4];
a.insert(1,4);
print(a);

# 리스트 요소 제거
a = [1,2,3,1,2,3];
a.remove(3);
print(a);

# pop() - 맨 마지막 인덱스 위치의 값을 리스트에서 빼낸다.
a = [1,2,3];
print(a.pop());
print(a);

b = [1,2,3,4,5];
print(b.pop(1)); # 1 => 인덱스번호
print(b);

# count() - 리스트에 포함된 특정 요소의 갯수 카운트
a = [1,2,3,1];
print(a.count(1));

# extend - 리스트의 확장 
a = [1,2,3];
a.extend([4,5]);
print(a);